### ISSUE TITLE
reactor.netty.channel.AbortedException: Connection has been closed

### ISSUE THREAD

**Issue by vaclap** [reactor/reactor-netty#2409]
It run well on reactor-netty which version belown 1.0.0, will occur error on 1.0.20
```
reactor.netty.channel.AbortedException: Connection has been closed
  at reactor.netty.http.server.HttpServerOperations.onInboundClose(HttpServerOperations.java:575) ~[reactor-netty-http-1.0.8.jar!/:1.0.8]
  Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException:
Error has been observed at the following site(s):
  |_ checkpoint ⇢ com.mt.jump.auth.AuthFilter [DefaultWebFilterChain]
  |_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
  |_ checkpoint ⇢ HTTP POST "/user/save" [ExceptionHandlingWebHandler]
Stack trace:
    at reactor.netty.http.server.HttpServerOperations.onInboundClose(HttpServerOperations.java:575) ~[reactor-netty-http-1.0.8.jar!/:1.0.8]
    at reactor.netty.channel.ChannelOperationsHandler.channelInactive(ChannelOperationsHandler.java:74) ~[reactor-netty-core-1.0.8.jar!/:1.0.8]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:241) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.CombinedChannelDuplexHandler$DelegatingChannelHandlerContext.fireChannelInactive(CombinedChannelDuplexHandler.java:418) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.handler.codec.ByteToMessageDecoder.channelInputClosed(ByteToMessageDecoder.java:389) ~[netty-codec-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.handler.codec.ByteToMessageDecoder.channelInactive(ByteToMessageDecoder.java:354) ~[netty-codec-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.CombinedChannelDuplexHandler.channelInactive(CombinedChannelDuplexHandler.java:221) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:241) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1405) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:262) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:248) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:901) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.AbstractChannel$AbstractUnsafe$8.run(AbstractChannel.java:831) ~[netty-transport-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) ~[netty-common-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) ~[netty-common-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:384) ~[netty-transport-native-epoll-4.1.65.Final-linux-x86_64.jar!/:4.1.65.Final]
    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) ~[netty-common-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.65.Final.jar!/:4.1.65.Final]
    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.1.65.Final.jar!/:4.1.65.Final]
    at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
```

* Reactor version(s) used: 1.0.20/1.0.19
* JVM version (`java -version`): jdk17

**Comment by violetagg** [comment]
@vaclap please try to provide a complete minimal sample (something that we can unzip or git clone and build) that reproduces the problem.

**Comment by vaclap** [comment]
> @vaclap please try to provide a complete minimal sample (something that we can unzip or git clone and build) that reproduces the problem.

I tried to reproduce this problem in development environment but failed，but when read reactor-netty-http source code , I found the difference between the two versions  the version less than 1.0.0 and 1.0.20.

1.0.20
reactor.netty.http.server.HttpServerOperations override the method onInboundClose.
```
@Override
	protected void onInboundClose() {
		discardWhenNoReceiver();
		if (!(isInboundCancelled() || isInboundDisposed())) {
			onInboundError(new AbortedException("Connection has been closed"));
		}
		terminate();
	}
```

but less than 1.0.0.
HttpServerOperations not override the onInboundClose method.

Can you tell me why this change is made？

**Comment by violetagg** [comment]
@vaclap This change was made in order to propagate to the server the information that the client closed the connection - see #1512

**Comment by vaclap** [comment]
> @vaclap This change was made in order to propagate to the server the information that the client closed the connection - see #1512

Thank you very much!
